lib/ui/semantics/attributed_string.cc

```diff
@@ -0,0 +1,105 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/semantics/attributed_string.h"
+
+#include "flutter/fml/logging.h"
+#include "flutter/lib/ui/ui_dart_state.h"
+#include "third_party/tonic/dart_args.h"
+#include "third_party/tonic/dart_binding_macros.h"
+
+#include <memory>
+
+namespace flutter {
+
+static void AttributedString_constructor(Dart_NativeArguments args) {
+  UIDartState::ThrowIfUIOperationsProhibited();
+  DartCallConstructor(&AttributedString::Create, args);
+}
+
+IMPLEMENT_WRAPPERTYPEINFO(ui, AttributedString);
+
+#define FOR_EACH_BINDING(V)                 \
+  V(AttributedString, initAttributedString) \
+  V(AttributedString, initWithConcat)       \
+  V(AttributedString, GetString)            \
+  V(AttributedString, toString)
+
+FOR_EACH_BINDING(DART_NATIVE_CALLBACK)
+
+void AttributedString::RegisterNatives(tonic::DartLibraryNatives* natives) {
+  natives->Register(
+      {{"AttributedString_constructor", AttributedString_constructor, 1, true},
+       FOR_EACH_BINDING(DART_REGISTER_NATIVE)});
+}
+
+AttributedString::AttributedString() {}
+
+AttributedString::~AttributedString() {}
+
+fml::RefPtr<AttributedString> AttributedString::Create() {
+  return fml::MakeRefCounted<AttributedString>();
+}
+
+void AttributedString::initAttributedString(
+    std::string string,
```

Comment:
Add a parameter here like `Dart_Handle dart_handle`. Use `AssociateWithDartWrapper` on the handle, and then you can then drop the constructor method.

Reply:
That will also require making this method static, like with `Vertices::init`.

I would move `string_` and `attributes_` to the ctor of the native object.